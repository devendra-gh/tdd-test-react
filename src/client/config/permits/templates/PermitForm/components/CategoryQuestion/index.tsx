import React from 'react';
import { IVariables } from '@tamm/app-composer';
// import Select from '@tamm/ui-lib-v2-select';
import FormTemplate from '@tamm/ui-lib-v2-form-template';

import PermitConfigs from '../../../../permitConfigs';

const permitCategories: any[] = [];
const permitConfigs: { [key: string]: any } = PermitConfigs;

Object.keys(PermitConfigs).forEach((permit: string) => {
  const {
    [permit]: { category },
  } = permitConfigs;

  permitCategories[category] = '';
});

const CategoryQuestion = (props: IVariables) => {
  // const { handleChange, selectedCategory, i18n, disablePermit } = props;

  // const items = Object.keys(permitCategories).map(category => ({
  //   id: category,
  //   label: i18n(`category.${category}`),
  // }));

  return (
    <React.Fragment>
      <FormTemplate
        formData={{}}
        // i18n={i18n}
        id="category"
        inputGroups={[
          {
            fields: [
              // {
              //   'aria-label':
              //     'Please select your permit type from drop-down list below',
              //   elementType: 'select',
              //   items,
              //   label: i18n('field.permitType.info'),
              //   name: 'select',
              //   value: selectedCategory,
              //   showSearch: false,
              //   onChange: handleChange,
              //   disabled: () => disablePermit,
              // },
            ],
            // name: i18n('title.selectPermitType'),
            // twoColumns: true,
          },
        ]}
      />
    </React.Fragment>
  );
};

export default CategoryQuestion;
