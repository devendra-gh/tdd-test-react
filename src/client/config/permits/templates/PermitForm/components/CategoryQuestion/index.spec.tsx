import React from 'react';
import { shallow } from 'enzyme';

import CategoryQuestion from './index';

// eslint-disable-next-line no-console
console.log = () => {};
console.info = () => {};
console.warn = () => {};
console.error = () => {};

describe('Category Question', () => {
  let props: any;

  beforeEach(() => {
    props = {
      i18n: jest.fn(),
      handleChange: jest.fn(),
    };
  });

  it('should render the category question successfully', () => {
    const component = shallow(<CategoryQuestion {...props} />);
    expect(component).toMatchSnapshot();
  });
});
