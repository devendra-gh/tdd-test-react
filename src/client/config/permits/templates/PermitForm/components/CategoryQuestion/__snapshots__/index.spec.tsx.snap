// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category Question should render the category question successfully 1`] = `
<Fragment>
  <LocalizedComponent(FormTemplate)
    cancelButton={null}
    cancelLink={null}
    formData={Object {}}
    i18n={[Function]}
    id="category"
    inputGroups={
      Array [
        Object {
          "fields": Array [],
        },
      ]
    }
    name={null}
    submitButton={null}
  />
</Fragment>
`;
